puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -190,17 +190,34 @@ def build_events_list(events)
   # @api private
   def resource_status_to_hash(resource_status)
     defaulted_corrective_change = defined?(resource_status.corrective_change) ? resource_status.corrective_change : nil
-    {
-      "skipped"           => resource_status.skipped,
-      "timestamp"         => Puppet::Util::Puppetdb.to_wire_time(resource_status.time),
-      "resource_type"     => resource_status.resource_type,
-      "resource_title"    => resource_status.title.to_s,
-      "file"              => resource_status.file,
-      "line"              => resource_status.line,
-      "containment_path"  => resource_status.containment_path,
-      "corrective_change" => defaulted_corrective_change,
-      "events"            => build_events_list(resource_status.events),
-    }
+    if resource_event_index_size_reached?(resource_status)
+      Puppet.warning("Skipping resource_event on #{self.host} caused by Resource type: #{resource_status.resource_type} in #{resource_status.file}:#{resource_status.line}")
+    else
+      {
+        "skipped"           => resource_status.skipped,
+        "timestamp"         => Puppet::Util::Puppetdb.to_wire_time(resource_status.time),
+        "resource_type"     => resource_status.resource_type,
+        "resource_title"    => resource_status.title.to_s,
+        "file"              => resource_status.file,
+        "line"              => resource_status.line,
+        "containment_path"  => resource_status.containment_path,
+        "corrective_change" => defaulted_corrective_change,
+        "events"            => build_events_list(resource_status.events),
+      }
+    end
+  end
+
+  # Check if resource_events_resource_timestamp_xxxxxz index size is reached
+  # resource_events_resource_timestamp_xxxxxz index is a multi column index that can fail if the
+  # fields are over 2704 bytes toghether
```

Comment:
resource_events_resource_timestamp_xxxxxz index is comprised of 3 columns (resource timestamp, title and type) and its maximum size is 2704 bytes. So the size of the 3 resource attributes combined, must not exceed 2704 bytes.

Reply:
Updated. Thanks!